File: packages/banxa_payments_flutter_android/android/src/main/kotlin/com/banxa/flutterpaymentsdk/PaymentMethodMapper.kt
package com.banxa.flutterpaymentsdk

/**
 * Maps Banxa partner-api `paymentMethodId` values to Primer type identifiers.
 *
 * Android Primer Checkout 3.x presents methods from the client session; the
 * mapped type is still useful for logging / future show-method APIs and for
 * parity with the iOS Drop-In path.
 */
object PaymentMethodMapper {
  fun primerType(banxaPaymentMethodId: String): String =
    when (banxaPaymentMethodId.lowercase()) {
      "debit-credit-card", "credit-card", "card", "primercc" -> "PAYMENT_CARD"
      "apple-pay" -> "APPLE_PAY"
      "google-pay" -> "GOOGLE_PAY"
      "paypal" -> "PAYPAL"
      "klarna", "klarna-paynow" -> "KLARNA"
      else -> banxaPaymentMethodId
    }
}

File: packages/banxa_payments_flutter_ios/ios/banxa_payments_flutter_ios/Sources/banxa_payments_flutter_ios/PaymentMethodMapper.swift
import Foundation

/// Maps Banxa partner-api `paymentMethodId` values to Primer Drop-In type strings.
enum PaymentMethodMapper {
  static func primerType(forBanxaPaymentMethodId id: String) -> String {
    switch id.lowercased() {
    case "debit-credit-card", "credit-card", "card", "primercc":
      return "PAYMENT_CARD"
    case "apple-pay":
      return "APPLE_PAY"
    case "google-pay":
      return "GOOGLE_PAY"
    case "paypal":
      return "PAYPAL"
    case "klarna", "klarna-paynow":
      return "KLARNA"
    default:
      // Pass through already-Primer values (e.g. PAYMENT_CARD).
      return id
    }
  }
}
